obs-packaging/gen_versions_txt.sh

```diff
@@ -77,7 +78,8 @@ kernel_version=${kernel_version}
 
 # Golang
 go_version=${golang_version}
-go_checksum=${golang_x84_64_sha256}
+go_checksum_x86_64=${golang_x84_64_sha256}
+go_checksum_ppc64le=${golang_ppc64le_sha256}
```

Comment:
Could you do the check on the arch here instead of in each `update.sh` files to keep the change localized?

Reply:
@marcov : Are you saying ... 
if arch is x86_64 then go_checksum=${golang_x84_64_sha256} else go_checksum=${golang_ppc64le_sha256}
Thus we need to no changes in update.sh ?
Also ` obs-packaging/versions.txt` will have only one `go_checksum` depending on the arch? Also, upstream file will have only one `go_checksum` containing the checksum of which arch?

Reply:
> > @marcov : Are you saying ...
> > if arch is x86_64 then go_checksum=${golang_x84_64_sha256} else go_checksum=${golang_ppc64le_sha256}
> > Thus we need to no changes in update.sh ?
> 
> Kind of.
> E.g. you may want to generate packaging files for ppc also when running on x86, as most developers do :)

@marcov : In that case, we cannot depend on $(uname -m). As we can see here in Point 6, we just run the script `https://github.com/kata-containers/packaging/blob/9c9aa22a6ab22ebf8d9c1e09f94bd9d7d85b4d40/release/release.md#start-release-process` . So do we pass the ARCH to it ? Example: ARCH=ppc64le ./gen_versions_txt.sh ${BRANCH} ?
With `ARCH="${ARCH:-amd64}" in gen_versions.txt.sh . `

> 
> > Also `obs-packaging/versions.txt` will have only one `go_checksum` depending on the arch?
> 
> No, you need to keep all checksums there to make the above work
> 
> > Also, upstream file will have only one `go_checksum` containing the checksum of which arch?
> 
> According to this PR, the checksum should match the architecture of the OBS project you are pushing to.

Reply:
I like @marcov  idea it should be good to push without depend the current architecure.

So I think we need to support a variable `ARCH` to honor if is provided.
 
```bash
ARCH=${ARCH:-$(uname -m)
```
I think for an easy path  we can define the golang_sha when source `pkglib.sh` 
```
ARCH=${ARCH:-$(uname -m)"
go_checksum=$(curl -s -L "https://storage.googleapis.com/golang/go${golang_version}.linux-${GOARCH}.tar.gz.sha256")
```
something like this may work for short therm, longer term we should refactor/re-rework the way the script that generate the spec/deb files work.